src/core/router/identity.cc

```diff
@@ -729,39 +727,37 @@ Keys CreateRandomKeys() {
 
 IdentHash CreateRoutingKey(
     const IdentHash& ident) {
-  std::uint8_t buf[41];  // ident + yyyymmdd
-  memcpy(buf, (const std::uint8_t *)ident, 32);
-  time_t t = time(nullptr);
-  struct tm tm;
-  // TODO(unassigned): never use sprintf, use snprintf instead.
-#ifdef _WIN32
-  gmtime_s(&tm, &t);
-  sprintf_s(
-      reinterpret_cast<char *>((buf + 32)),
-      9,
-      "%04i%02i%02i",
-      tm.tm_year + 1900,
-      tm.tm_mon + 1,
-      tm.tm_mday);
-#else
-  gmtime_r(&t, &tm);
-  sprintf(
-      reinterpret_cast<char *>((buf + 32)),
-      "%04i%02i%02i",
-      tm.tm_year + 1900,
-      tm.tm_mon + 1,
-      tm.tm_mday);
-#endif
   IdentHash key;
-  // TODO(anonimal): this try block should be larger or handled entirely by caller
-  try {
-    kovri::core::SHA256().CalculateDigest((std::uint8_t *)key, buf, 40);
-  } catch (...) {
-    core::Exception ex;
-    ex.Dispatch(__func__);
-    // TODO(anonimal): review if we need to safely break control, ensure exception handling by callers
-    throw;
-  }
+  if (ident.IsZero())
+    {
+      LOG(error) << __func__
+                 << ": cannot create routing key from zero-initialized ident";
+      return key;
```

Comment:
We don't want to return an object simply to satisfy a return type. `IsZero()` was a hack to work with the union and should be eliminated / not built upon. We can keep IsZero for now but we should throw and deal with consequences in any callers.

Reply:
Alright, that makes sense. I'll try to think of alternatives too (for another PR).